### ISSUE TITLE
Tavern invite changes the previously selected hero

### ISSUE THREAD

**Issue by Olegmods** [vcmi/vcmi#4723]
**Describe the bug**
When you choose a hero in tavern invite window for the first time, then exit the tavern. When you enter the tavern, the hero to invite is different

**Game logs**
Not needed

**To Reproduce**
1. Open the tavern on the first day
2. Select a hero to invite
3. Exit the tavern
4. Return to the tavern
5. See error (the hero for invite isn't the one you selected)

**Expected behavior**
The selected hero stays.

**Actual behavior**
Selected hero changes to another hero.

**Did it work earlier?**
I don't think so.

**Screenshots**
Even a video! Link:
https://vk.com/doc819525685_680274117?hash=sTD3s6c3Iv8zgrF32C8278ahgcu3pXfdZnfA8TGfBjs&dl=nrZlYMiAzw0oOJZMaIKxre9yZyN4vr2IeB3BQGwzg7H

**Version**
 - OS: Windows, Android
 - Version: 1.5.7

**Additional context**
@Laserlicht is this normal? And if it isn't, please fix it.

**Comment by Laserlicht** [comment]
Yes. The selected Hero isn't saved. That's also the case for HD mod (the inspiration source). Difference is only that HD mod has a separate image for "random" by default. VCMI just selects a random hero.

**Comment by Olegmods** [comment]
> Yes. The selected Hero isn't saved. That's also the case for HD mod (the inspiration source). Difference is only that HD mod has a separate image for "random" by default. VCMI just selects a random hero.
> 
> I'm not sure if we need a fix for this. Maybe I would change it also like in hd mod to make it clearer.

Ok. I haven't used HD mod for a long time, so I don't remember how it was there. But it will be better if every hero selected were saved and would change only next week. It feels more clear, maybe.

Should I close the issue then?

**Comment by Laserlicht** [comment]
My PR does not fully address it. But I will not add a saving at the moment, as this might even have to be mirrored on the server.

**Comment by Olegmods** [comment]
> My PR does not fully address it. But I will not add a saving at the moment, as this might even have to be mirrored on the server.
> 
> So feel free to leave it open if you want.

Ok, then I'll leave that open. Hope saving can be done.
